main.py

```diff
@@ -1,29 +1,84 @@
 """
 This is an Add-On for DocumentCloud which wraps the tabula-py library
 https://github.com/chezou/tabula-py
-
-It attempts to convert tables from PDFs to CSVs
+It allows you to export tables from a set of PDFs into CSVs and
+download the resulting CSVs as a zip file.
 """
 
+import os
+import sys
+import time
+import fnmatch
+import zipfile
 import tabula
 from documentcloud.addon import AddOn
+from clouddl import grab
 
 
 class Tabula(AddOn):
-    """A tabula Add-On for DocumentCloud"""
+    """A tabula PDF table extraction Add-On for DocumentCloud"""
 
-    def main(self):
+    def fetch_template(self, url):
+        """Fetch the template from either Dropbox or Google Drive"""
+        os.makedirs(os.path.dirname("./out/"), exist_ok=True)
+        if grab(url, "./out/"):
+            for file in os.listdir("./out/"):
+                if fnmatch.fnmatch(file, "*.json"):
+                    template_path = os.path.join("./out/", file)
+                    os.rename(template_path, os.path.join("./out/", "template.json"))
+                    return True
+        self.set_message(
+            "No valid JSON tabula template was found in the URL provided, exiting..."
+        )
+        time.sleep(3)
```

Comment:
No reason to sleep here, the otuput message is saved in the database and will be shown to the user